venter/serializers.py

```diff
@@ -33,7 +34,7 @@ class Meta:
         model = Complaints
         fields = (
             'id', 'created_by', 'description', 'report_date', 'status', 'latitude',
-            'longitude', 'location_description', 'tags', 'comments', 'users_up_voted', 'images'
+            'longitude', 'location_description', 'tags', 'comments', 'users_up_voted', 'subscriptions', 'images'
```

Comment:
I'd like to know the underlying logic there. I don't think I get what you're passing to the functions. Could you please explain what you're doing in user_ues in the serializer file?

Reply:
`user_ues` is defined as a method field on line 43 and is assigned on line 44 (sorry for the terrible naming)
https://github.com/wncc/IITBapp/blob/2b6a3b31b93a9fccdcfc91fc53719cc63667a066/events/serializers.py#L43

The actual function is a static method defined at line 14. `slf` refers to `self` i.e. the serializer class' instance and `obj` refers to the `Event` object. The terrible way I've done this isn't too inefficient cause everything is already preloaded anyway, but there's probably a much cleaner way to do this

https://github.com/wncc/IITBapp/blob/2b6a3b31b93a9fccdcfc91fc53719cc63667a066/events/serializers.py#L14

Reply:
FYI, I've just refactored this in #311, so the links above won't show the latest (cleaner) version.

Reply:
Woah, this is a bit confusing now.
> The API endpoint for listing all complaints is different compared to the one for retrieving a particular complaint.

This is just RESTful design

>  If I force the complaints list to show subscription status instead of subscribers list, then we won't be able to see the subscribers for the complaints contained within the list.

That's the point. Why do we _want_ to see the subscriber list? Nobody really cares who is subscribed to which complaint except the logged in user. It's just going to add to the serialization and network overhead.

> What I can do instead is tweak the routing and views such that for each complaint that is retrieved individually (i.e using its pk), I can use validation logic to check whether the user is currently subscribed to the complaint in question.

Yes, you'll need to do this for _both_, the list and individual complaint endpoints, if that's what you mean to say. Again, we don't need the list of subscribers in the final JSON response.

> Removing the list of all subscribers from the complaint response would be valid if it were being retrieved for an individual user.

It is being retrieved for an individual user. That's the whole point of the login thing.

>  I think it is necessary for a generic response which is supposed to contain details about all the complaints, since the subscription list is supposed to track all the users following these complaints.

Yes, but putting everything in the JSON response would basically mean downloading half the database every time you get the list of complaints. Don't forget you're making one db call for every user, so this is going to be crazy slow.

> The end result of this, would be that I could try tweaking the response for retrieving individual complaints. As it stands, the request is being pulled and passed to the serializer as extra context when a single complaint is being retrieved. I could use this request to perform subscription validation for each complaint individually in its individual response while keeping the subscription list intact for the Complaint List response (which is what the above serialiser with the subscription list will be used for).

Again, you'll need to do that for individual responses as well, but this subscription list is just terrible for performance. It's great to have all the data at once on the client, but neither is it something that a $5 server can do, nor something that's good design.

Reply:
Cool. That makes more sense now. Thanks